data/media/src/main/kotlin/com/google/jetpackcamera/data/media/MediaRepository.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera.data.media
+
+import android.graphics.Bitmap
+import android.net.Uri
+
+/**
+ * Data layer for Media.
+ */
+interface MediaRepository {
+    suspend fun getLastCapturedMedia(): MediaDescriptor
+    suspend fun load(mediaDescriptor: MediaDescriptor): Media
+}
+
+/**
+ * Descriptors used for [Media].
+ *
+ * Media descriptors contain a reference to a [Media] item that's not yet loaded.
+ */
+sealed interface MediaDescriptor {
+    data object None : MediaDescriptor
+    class Image(val uri: Uri, val thumbnail: Bitmap?) : MediaDescriptor
+    class Video(val uri: Uri, val thumbnail: Bitmap?) : MediaDescriptor
```

Comment:
since they have the same properties, would it make sense to wrap `Image` and `Video` in another interface that extends `MediaDescriptor`?

I think this should solve the smart cast issue you mentioned in [this thread](https://github.com/google/jetpack-camera-app/pull/330#discussion_r2001583391) for #330 

```suggestion
    sealed interface Available : MediaDescriptor {
        val uri: Uri
        val thumbnail: Bitmap?
        class Image(override val uri: Uri, override val thumbnail: Bitmap?) : Available
        class Video(override val uri: Uri, override val thumbnail: Bitmap?) : Available
    }
```

Reply:
While it would help with the smart cast, I don't think it would be worth it with the added complexity introduced.